internal/user/user_profile_hash_v1.go

```diff
@@ -10,31 +10,35 @@ import (
 	"github.com/kopia/kopia/internal/crypto"
 )
 
-// parameters for v1 hashing.
-const (
-	hashVersion1 = 1
-
-	v1SaltLength = 32
-)
-
 //nolint:gochecknoglobals
-var dummyV1HashThatNeverMatchesAnyPassword = make([]byte, crypto.MasterKeyLength+v1SaltLength)
+var dummyV1HashThatNeverMatchesAnyPassword = make([]byte, crypto.MasterKeyLength+crypto.V1SaltLength)
+
+func (p *Profile) setPassword(password string) error {
+	if len(p.KeyDerivationAlgorithm) == 0 {
+		if p.PasswordHashVersion == 0 {
+			return errors.New("key derivation algorithm and password hash version not set")
+		}
+		// Setup to handle legacy hashVersion.
+		if p.PasswordHashVersion == crypto.HashVersion1 {
+			p.KeyDerivationAlgorithm = crypto.ScryptAlgorithm
```

Comment:
So, are you trying to use this as migration away from the old `PasswordHashVersion` ? (set new field and hope the caller saves the new profile)

I think the best way to do this migration is through an explicit "format upgrade" track / a command or something that customers have to run on their repository to move away from deprecated fields, otherwise we'll never be "sure" if this has already been done or not because we do not know whether everyone has reset their passwords using the new profile fields.

If that is the case, then do you think `p.KeyDerivationAlgorithm` should be set here ? If it SHOULD be set then why not also unset `p.PasswordHashVersion` ?